src/codegen/replacements.ts

```diff
@@ -1,29 +1,7 @@
 import { LoaderKeys } from "../api/schema";
 import { sliceSourceCode } from "./builtin-parser";
 
-// This is a list of extra syntax replacements to do. Kind of like macros
-// These are only run on code itself, not string contents or comments.
-export const replacements: ReplacementRule[] = [
-  { from: /\bthrow new TypeError\b/g, to: "$throwTypeError" },
-  { from: /\bthrow new RangeError\b/g, to: "$throwRangeError" },
-  { from: /\bthrow new OutOfMemoryError\b/g, to: "$throwOutOfMemoryError" },
-  { from: /\bnew TypeError\b/g, to: "$makeTypeError" },
-  { from: /\bexport\s*default/g, to: "$exports =" },
-];
-
-// These rules are run on the entire file, including within strings.
-export const globalReplacements: ReplacementRule[] = [
-  {
-    from: /\bnotImplementedIssue\(\s*([0-9]+)\s*,\s*((?:"[^"]*"|'[^']+'))\s*\)/g,
-    to: "new TypeError(`${$2} is not implemented yet. See https://github.com/oven-sh/bun/issues/$1`)",
-  },
-  {
-    from: /\bnotImplementedIssueFn\(\s*([0-9]+)\s*,\s*((?:"[^"]*"|'[^']+'))\s*\)/g,
-    to: "() => $throwTypeError(`${$2} is not implemented yet. See https://github.com/oven-sh/bun/issues/$1`)",
-  },
```

Comment:
Before this PR the `globalReplacements` `.to` was being transformed with`.replaceAll("$", "__intrinsic__")` which would have made the `$2 and `$1` incorrectly transformed because it didn't have the transform that the `replacements` `.to` has for `%1`, `%2`, etc.